runtime/compiler/optimizer/J9Simplifier.cpp

```diff
@@ -479,28 +479,73 @@ TR::Node *
 J9::Simplifier::simplifyaCallMethods(TR::Node * node, TR::Block * block)
    {
    if ((node->getOpCode().isCallDirect()) && !node->getSymbolReference()->isUnresolved() &&
-       (node->getSymbol()->isResolvedMethod()) &&
-       ((node->getSymbol()->getResolvedMethodSymbol()->getRecognizedMethod() == TR::java_math_BigDecimal_valueOf) ||
-        (node->getSymbol()->getResolvedMethodSymbol()->getRecognizedMethod() == TR::java_math_BigDecimal_add) ||
-        (node->getSymbol()->getResolvedMethodSymbol()->getRecognizedMethod() == TR::java_math_BigDecimal_subtract) ||
-        (node->getSymbol()->getResolvedMethodSymbol()->getRecognizedMethod() == TR::java_math_BigDecimal_multiply) ||
-        (node->getSymbol()->getResolvedMethodSymbol()->getRecognizedMethod() == TR::java_math_BigInteger_add) ||
-        (node->getSymbol()->getResolvedMethodSymbol()->getRecognizedMethod() == TR::java_math_BigInteger_subtract) ||
-        (node->getSymbol()->getResolvedMethodSymbol()->getRecognizedMethod() == TR::java_math_BigInteger_multiply)) &&
-       (node->getReferenceCount() == 1) &&
-       performTransformation(comp(), "%sRemoved dead BigDecimal/BigInteger call node [" POINTER_PRINTF_FORMAT "]\n", optDetailString(), node))
+       (node->getSymbol()->isResolvedMethod()))
       {
-      TR::Node *firstChild = node->getFirstChild();
-      anchorChildren(node, _curTree);
+      bool isSimplifiableMethod = false;
+      bool isIntOrDecimalNonDivArithOp = false;
```

Comment:
this is more rightly called isInstanceMethod since instance methods are the ones which need to NULLCHK the receiver.

Reply:
@andrewcraik Andrew, as we discussed, I changed the name to `requiresArgNULLCHK` to clarify why we're tracking these particular methods.  Fixed in ed97fb6168f0220071a8694fe1b19cc919d3af73

If you're OK with this change, I can squash the two commits prior to delivery.